tests/test_zuora_interrupted_sync.py

```diff
@@ -0,0 +1,177 @@
+        return "tap_tester_zuora_interrupted_test"
+
+    def test_run(self):
+        """ Executing tap-tester scenarios for both types of zuora APIs AQUA and REST"""
+        # Testing for only AQUA mode to reduce the execution time
+        self.run_test("AQUA")
+
+    def run_test(self, api_type):
+        """
+        Scenario: A sync job is interrupted. The state is saved with `current_stream`.
+                  The next sync job kicks off and the tap picks back up on that `current_stream` stream.
+        Expected State Structure:
+            {
+                "current_stream": "stream-name",
+                "bookmarks": {
+                    "stream-name-1": "bookmark-date"
+                    "stream-name-2": "bookmark-date"
+                }
+            }
+        Test Cases:
+        - Verify an interrupted sync can resume based on the `current_stream` and stream level bookmark value.
+        - Verify only records with replication-key values greater than or equal to the stream level bookmark are
+            replicated on the resuming sync for the interrupted stream.
+        - Verify the yet-to-be-synced streams are replicated following the interrupted stream in the resuming sync.
+        """
+        self.zuora_api_type = api_type
+        self.start_date = dt.strftime(utils.now() - timedelta(days=10), "%Y-%m-%dT00:00:00Z")
+        start_date_datetime = dt.strptime(self.start_date, "%Y-%m-%dT%H:%M:%SZ")
+        expected_streams = {'PaymentMethodTransactionLog'}
+
+        conn_id = connections.ensure_connection(self, original_properties=False)
+
+        # Run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Catalog selection
+        for catalog in found_catalogs:
+            if catalog["stream_name"] not in expected_streams:
+                continue
+
+            annoted_schema = menagerie.get_annotated_schema(conn_id, catalog["stream_id"])
+
+            # De-select all fields
+            non_selected_properties = annoted_schema.get("annotated-schema", {}).get("properties", {})
+            non_selected_properties = non_selected_properties.keys()
+            additional_md = []
+            connections.select_catalog_and_fields_via_metadata(conn_id,catalog,annoted_schema,additional_md=additional_md,non_selected_fields=non_selected_properties)
+
+        # Run sync
+        record_count_by_stream_full_sync = self.run_and_verify_sync(conn_id)
+        synced_records_full_sync = runner.get_records_from_target_output()
+        full_sync_state = menagerie.get_state(conn_id)
+
+        # State to run 2nd sync
+        #   orders: currently syncing
+        #   charges: synced records successfully
+        #   subscriptions: remaining to sync
+        state = {
+            "current_stream": "OrderAction",
+            "bookmarks":{'RevenueEventItem': {'version': 1665422431, 'UpdatedDate': dt.strftime(utils.now() - timedelta(days=5), "%Y-%m-%dT00:00:00Z")}}
+        }
```

Comment:
Format dict to reduce the line length.

Reply:
On the contrary, IMO, we can simply use `calculated_states_by_stream()` method in `base.py` to calculate the interrupted sync.

Reply:
made the changes